app/views/courses/show.html.erb

```diff
@@ -0,0 +1,28 @@
+<h1><%= @course.display_name %></h1>
+<hr>
+<% if current_user.nil? %>
+<div>You need to sign in to be able to join this class.</div>
+<br>
+<div class="sign-in-panel">
+<% if Rails.env == "development" %>
+<%= link_to "Developer Login", "/home/developer_login" %><br />
+<% end %>
+<%= render "devise/shared/links" %>
```

Comment:
Like where?  I'd really rather have users get directed to a sign up page, because then we can keep a unified course loading system, and not duplicate code.